### PR TITLE
minor changes + stage-specific event checks

### PR DESCRIPTION
**Pull request by frantastic24**
- section lua variables are properly set from start
- remove a duplicate lua function
- easily convert an RGB color to Hex through lua
- colorSwap lua support
- fix text being pasted twice
- stage-specific events will not crash by simply not loading any assets or triggering at all if used incorrectly. cope. seethe.

### CODE REVIEW

File: source/FunkinLua.hx [hunk 1]
```diff
@@ -1002,20 +1002,6 @@ class FunkinLua {
 			}
 			return boobs;
 		});
-		Lua_helper.add_callback(lua, "noteTweenAngle", function(tag:String, note:Int, value:Dynamic, duration:Float, ease:String) {
```
Comment (Itz-Miles) [inline source/FunkinLua.hx:1005]: I haven't caught up with 0.6+ Funkin' Lua, is there any reason why this callback is omitted?

**Review by GO-Miles — APPROVED** [review]
porpoise, quite possibly

**Review by iym1337 — APPROVED** [review]

File: source/FunkinLua.hx (hunk 1 above)
Comment (frantastic24) [reply source/FunkinLua.hx:1005]: Yes
> remove a duplicate lua function

**Review by ManMaster707 — APPROVED** [review]

**Review by stilic — APPROVED** [review]

**Review by CrusherNotDrip — APPROVED** [review]

**Review by TheKitBoi — APPROVED** [review]

**Review by Raltyro — APPROVED** [review]

File: source/FunkinLua.hx (hunk 1 above)
Comment (ImaginationSuperHero528) [reply source/FunkinLua.hx:1005]: Since when is this a duplicate?

File: source/PlayState.hx [hunk 2]
```diff
@@ -2455,61 +2455,64 @@ class PlayState extends MusicBeatState
...
-				dadbattleSmokes.add(smoke);
-				var smoke:BGSprite = new BGSprite('smoke', 1550 + offsetX, 660 + FlxG.random.float(-20, 20), 1.2, 1.05);
-				smoke.setGraphicSize(Std.int(smoke.width * FlxG.random.float(1.1, 1.22)));
-				smoke.updateHitbox();
-				smoke.velocity.x = FlxG.random.float(-15, -22);
-				smoke.active = true;
-				smoke.flipX = true;
-				dadbattleSmokes.add(smoke);
-
+				if (WeekData.getWeekFileName() == 'week1' && curStage == 'stage') { // PLEASE SHUT THE FUCK UP
```
Comment (ImaginationSuperHero528) [inline source/PlayState.hx:2458]: You should probably make it just the stage.

File: source/PlayState.hx [hunk 3]
```diff
@@ -2455,61 +2455,64 @@ class PlayState extends MusicBeatState
...
-				phillyGlowGradient.visible = false;
-				insert(members.indexOf(blammedLightsBlack) + 1, phillyGlowGradient);
-				if(!ClientPrefs.flashing) phillyGlowGradient.intendedAlpha = 0.7;
-
-				precacheList.set('philly/particle', 'image'); //precache particle image
-				phillyGlowParticles = new FlxTypedGroup<PhillyGlow.PhillyGlowParticle>();
-				phillyGlowParticles.visible = false;
-				insert(members.indexOf(phillyGlowGradient) + 1, phillyGlowParticles);
-		}
+				if (WeekData.getWeekFileName() == 'week3' && curStage == 'philly') { // PLEASE SHUT THE FUCK UP
```
Comment (ImaginationSuperHero528) [inline source/PlayState.hx:2493]: Again... the stage.

File: source/PlayState.hx [hunk 4]
```diff
@@ -3345,15 +3350,16 @@ class PlayState extends MusicBeatState
...
-				var lightId:Int = Std.parseInt(value1);
-				if(Math.isNaN(lightId)) lightId = 0;
-
-				var doFlash:Void->Void = function() {
-					var color:FlxColor = FlxColor.WHITE;
-					if(!ClientPrefs.flashing) color.alphaFloat = 0.5;
-
-					FlxG.camera.flash(color, 0.15, null, true);
-				};
+				if (WeekData.getWeekFileName() == 'week3' && curStage == 'philly') {
```
Comment (ImaginationSuperHero528) [inline source/PlayState.hx:3353]: I'm just gonna comment on each of these lines.

**Review by rodney528 — APPROVED** [review]

File: source/FunkinLua.hx (hunk 1 above)
Comment (frantastic24) [reply source/FunkinLua.hx:1005]: idk since a couple of months at least

File: source/PlayState.hx (hunk 2 above)
Comment (frantastic24) [reply source/PlayState.hx:2458]: No

File: source/PlayState.hx (hunk 4 above)
Comment (frantastic24) [reply source/PlayState.hx:3353]: I'm just gonna take your vital organs

**Review by rodney528 — APPROVED** [review]

**Review by DillyzThe1 — APPROVED** [review]

**Review by MemeHoovy — APPROVED** [review]

File: source/flixel/addons/ui/FlxInputText.hx [hunk 5]
```diff
@@ -480,7 +480,7 @@ class FlxInputText extends FlxText
 	/**
 	 * Inserts a substring into a string at a specific index
 	 *
-	 * @param	Insert			The string to have something inserted into
+	 * @param	Original			The string to have something inserted into
```
Comment (l1ttleO) [inline source/flixel/addons/ui/FlxInputText.hx:483]:
```suggestion
	 * @param	Original		The string to have something inserted into

```

Comment (frantastic24) [reply source/flixel/addons/ui/FlxInputText.hx:483]: What

Comment (l1ttleO) [reply source/flixel/addons/ui/FlxInputText.hx:483]: you have wrong indenting

Comment (frantastic24) [reply source/flixel/addons/ui/FlxInputText.hx:483]: Oh no! Anyways.

**Review by rodney528 — APPROVED** [review]
@ShadowMario should really approve this!

**Review by Intancote — APPROVED** [review]

**Review by Parsellol — APPROVED** [review]

**Review by rodney528 — APPROVED** [review]

**Pull request closed without merging**